### PR TITLE
BLD: Conda Recipe and CI

### PR DESCRIPTION
**Pull request by ZLLentz**
Main Goals:
- Reorganize module inside an inner `pmgr` directory
- Add `setup.py`
- Use `versioneer`
- Add `conda-recipe`
- Add `.travis.yml` that just builds and uploads

Consequences:
- I had to change all the in-package imports to be relative imports, or else it wouldn't work properly in the distribution
- The old launcher scripts no longer work, or really even make sense. I have removed them. This means that pmgr is tied to a conda environment intimately: the links in the apps directory should point to the `conda_wrappers` directory in the `bin` of the relevant conda environment.

### CODE REVIEW

File: conda-recipe/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+requirements:
+  build:
+    - python {{PY_VER}}*,>=3.6
+    - setuptools
+
+  run:
+    - python {{PY_VER}}*,>=3.6
+    - pyqt =5.6
+    - pyca >=3
+    - pykerberos >=1.1.14
```
Comment (teddyrendahl) [inline conda-recipe/meta.yaml]: Saw note in commit. This is the same package used by the `elog` so should be fine.

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import versioneer
+from setuptools import setup, find_packages
+
+setup(name='pmgr',
+      version=versioneer.get_version(),
+      cmdclass=versioneer.get_cmdclass(),
+      license='BSD',
+      author='SLAC National Accelerator Laboratories',
```
Comment (teddyrendahl) [inline setup.py]: Plural?

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+      packages=find_packages(),
+      include_package_data=True,
+      description=('Parameter Manager for LCLS Device Configurations'),
+      entry_points={'console_scripts': ['pmgr = pmgr.pmgr:main',
```
Comment (teddyrendahl) [inline setup.py]: What does `entry_points={console_scripts:  ...}` do as opposed to just `scripts=[...]`?

Comment (ZLLentz) [reply setup.py]: `entry_points={console_scripts: ...}` automatically generates a script that calls a function. `scripts` includes things that are marked as executables. In this case, the existing `pmgr`, etc. scripts were fully setting up a conda environment from a fixed location on NFS, which no longer makes sense if this is itself a conda package.

**Pull request merged**